osu.Game/Scoring/ScoreManager.cs

```diff
@@ -150,7 +158,7 @@ public async Task<long> GetTotalScoreAsync([NotNull] ScoreInfo score, ScoringMod
                     beatmapMaxCombo = score.BeatmapInfo.MaxCombo.Value;
                 else
                 {
-                    if (score.BeatmapInfo.ID == 0 || difficulties == null)
+                    if (!score.BeatmapInfo.IsManaged || difficulties == null)
```

Comment:
https://github.com/ppy/osu/pull/16428#issuecomment-1013687921 is caused by this change. This used to mean "this beatmap is available locally" which is not the same as `IsManaged` because of detaching. Requerying seems out of the question and I'm not sure where to take this. Seems that best that can be done is a `MaxCombo == 0` check after diffcalc is done but that feels dicey.

Reply:
Hmm, `MaxCombo` is actually nullable for some reason, so checking whether null or not should in theory be enough?

Reply:
It's not nullable on the `StarDifficulty` that diffcalc returns if I remember correctly, I was thinking to check that one.

Reply:
It looks like this conditional can be removed completely these days. I'll push this for some further testing, but it looks to work as expected locally in all the scenarios I can find.